core/src/main/java/org/infinispan/marshall/core/Ids.java

```diff
@@ -9,178 +9,7 @@
-
-   int FLAG = 85;
-
-   int STATE_CHUNK = 86;
-   int CACHE_JOIN_INFO = 87;
-
-   /* 88-90 reserved in org.infinispan.commons.marshall.Ids */
-
-   int DEFAULT_CONSISTENT_HASH_FACTORY = 91;
-   int REPLICATED_CONSISTENT_HASH_FACTORY = 92;
-   int SYNC_CONSISTENT_HASH_FACTORY = 93;
-   int TOPOLOGY_AWARE_CONSISTENT_HASH_FACTORY = 94;
-   int TOPOLOGY_AWARE_SYNC_CONSISTENT_HASH_FACTORY = 95;
-   int SIMPLE_CLUSTERED_VERSION = 96;
-   int DELTA_COMPOSITE_KEY = 97;
-
-   int EMBEDDED_METADATA = 98;
-
-   int NUMERIC_VERSION = 99;
-   int SCOPED_KEY = 100;
-
-   int NON_EXISTING_VERSION = 101;
-
-   int CACHE_NOT_FOUND_RESPONSE = 102;
-   int KEY_VALUE_PAIR_ID = 103;
-   int INTERNAL_METADATA_ID = 104;
-   int MARSHALLED_ENTRY_ID = 105;
-
-   /* 106 -120 reserved in org.infinispan.commons.marshall.Ids */
-
-   int ENUM_SET_ID = 121;
-   int LIST_ARRAY = 122;
-
-   int SIMPLE_COLLECTION_KEY_FILTER = 123;
-   int KEY_FILTER_AS_KEY_VALUE_FILTER = 124;
-   int CLUSTER_EVENT = 125;
-   int CLUSTER_LISTENER_REMOVE_CALLABLE = 126;
-   int CLUSTER_LISTENER_REPLICATE_CALLABLE = 127;
-   int CLUSTER_EVENT_CALLABLE = 128;
-   int X_SITE_STATE = 129;
-   int COMPOSITE_KEY_VALUE_FILTER = 130;
-   int DELTA_MAPREDUCE_LIST_ID = 131;
-   int DELTA_AWARE_MAPREDUCE_LIST_ID = 132;
-   int CACHE_STATUS_RESPONSE = 133;
-   int CACHE_EVENT_CONVERTER_AS_CONVERTER = 134;
-   int CACHE_EVENT_FILTER_AS_KEY_VALUE_FILTER = 135;
-   int CONVERTER_AS_CACHE_EVENT_CONVERTER = 136;
-   int KEY_FILTER_AS_CACHE_EVENT_FILTER = 137;
-   int KEY_VALUE_FILTER_AS_CACHE_EVENT_FILTER = 138;
-   int NULL_VALUE_CONVERTER = 139;
-   int ACCEPT_ALL_KEY_VALUE_FILTER = 140;
-   int MANAGER_STATUS_RESPONSE = 141;
-   int MULTI_CLUSTER_EVENT_CALLABLE = 142;
-   int COMPOSITE_KEY_FILTER = 143;
-   int KEY_VALUE_FILTER_AS_KEY_FILTER = 144;
-   int CACHE_EVENT_FILTER_CONVERTER_AS_KEY_VALUE_FILTER_CONVERTER = 145;
-
-   int INTERMEDIATE_OPERATIONS = 146;
-   int TERMINAL_OPERATIONS = 147;
-   int STREAM_MARSHALLING = 148;
-
-   int COMMAND_INVOCATION_ID = 149;
-   int CACHE_FILTERS = 150;
-
-   int OPTIONAL = 151;
-
-   // Functional
-   int META_PARAMS_INTERNAL_METADATA = 152;
-   int META_PARAMS = 153;
-
-   // TODO: Add other meta params
-   int META_LIFESPAN = 154;
-   int META_ENTRY_VERSION = 155;
-   int NUMERIC_ENTRY_VERSION = 156;
-
-   int READ_WRITE_SNAPSHOT_VIEW = 157;
-
-   // 158 - 160 used by functions in commons
-   int AVAILABILITY_MODE = 161;
-   int VALUE_MATCHER = 162;
-
-   int HASH_FUNCTION_PARTITIONER = 163;
-   int SYNC_REPLICATED_CONSISTENT_HASH_FACTORY = 164;
-
-   int AFFINITY_FUNCTION_PARTITIONER = 165;
-
-   int PERSISTENT_UUID = 166;
-
-   // 167 is ImmutableEntry in commons
-   int COLLECTIONS = 168;
-
-   int READ_ONLY_SNAPSHOT_VIEW = 169;
-   int NO_VALUE_READ_ONLY_VIEW = 170;
-
-   int MIME_CACHE_ENTRY = 171;
-   int UUID = 172;
-
-   // 173-174 used in commons
-   int ARRAYS = 175;
+   // All identifiers now defined in commons for a more complete view.
```

Comment:
It would created a lot of noise in this PR, since you'd have to change all references to this class. Can be done at a later stage when cleaning things up. I can probably do it in PR 3/3 where I'm adding ExternalPojo marker to mark classes that are expected to be externally marshallable (see Dan's comment in previous PR).